R/zchunk_LB132.ag_an_For_Prices_USA_C_2005.R

```diff
@@ -38,43 +39,167 @@ module_aglu_LB132.ag_an_For_Prices_USA_C_2005_DISABLED <- function(command, ...)
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
+    # Converting cotton back to primary equivalent (seed cotton)
+    # Seed cotton has no price in PRICESTAT. Need to derive its price from cotton lint and cottonseed
+    # Assigning a price for game meat so that OtherMeat is assigned a price
+    FAO_USA_ag_an_P_USDt_PRICESTAT %>%
+      select(-country.codes, -item.codes, -element, -element.codes) %>%
+      gather(year, price, -countries, -item) %>%
+      filter(item == "Seed cotton" | item == "Cotton lint" | item == "Cottonseed" | item == "Game meat" | item == "Cattle meat") %>%
+      mutate(item = sub(" ", "_", item)) %>%
+      spread(item, price) %>%
+      mutate(Seed_cotton = Cotton_lint * CONV_COTTON_LINT + Cottonseed * (1 - CONV_COTTON_LINT)) %>%
+      mutate(Game_meat = Cattle_meat) %>%
+      gather(item, price, -countries, -year) %>%
+      mutate(item = sub("_", " ", item)) ->
+      extra_price
+    # Put these calculated prices back to the dataset
+    FAO_USA_ag_an_P_USDt_PRICESTAT %>%
+      select(-country.codes, -item.codes, -element, -element.codes) %>%
+      gather(year, price, -countries, -item) %>%
+      filter(!(item == "Seed cotton" | item == "Cotton lint" | item == "Cottonseed" | item == "Game meat" | item == "Cattle meat")) %>%
+      bind_rows(extra_price) %>%
+      # Calculate unweighted averages for each FAO commodity over price years
+      filter(year %in% MODEL_PRICE_YEARS) %>%
+      group_by(countries, item) %>%
+      summarise_at(vars(price), mean, na.rm = T) %>%
+      ungroup() %>%
+      filter(!is.na(price)) ->
+      Avg_price
+
+    # Computing average prices and production quantities of all commodities
+    # Subset only the relevant country/item combinations from the ag prodstat database
+    # NOTE: This mostly excludes fodder crops and "not elsewhere specified" crops. It also excludes about seven crops that are
+    # minor enough to not distort price estimates, but their omission may cause problems if a large number of commodities is used
+
+    # Build tables with production and price, and calculate revenue
+    # Part 1: Primary agricultural goods and animal products
+    # Primary agricultural goods
+    L100.FAO_ag_Prod_t %>%
+      filter(year %in% MODEL_PRICE_YEARS) %>%
+      rename(prod = value) %>%
+      group_by(countries, item) %>%
+      summarise_at(vars(prod), mean, na.rm = T) %>%
+      ungroup() %>%
+      filter(!is.na(prod)) %>%
+      # Calculating revenue by commodity as production times price
+      inner_join(Avg_price, by = c("countries", "item")) %>%
+      mutate(V_USD = price * prod) %>%
+      left_join_error_no_match(FAO_ag_items_PRODSTAT[c("item", "GCAM_commodity")], by = "item") %>%
```

Comment:
Change to `left_join_error_no_match(select(FAO_ag_items_PRODSTAT, item, GCAM_commodity)), by = "item")` for consistency?